examples/generative/molecule_generation.py

```diff
@@ -374,12 +379,16 @@ def get_decoder(dense_units, dropout_rate, latent_dim, adjacency_shape, feature_
 
 
 class Sampling(layers.Layer):
+    def __init__(self, seed=None, **kwargs):
+        super().__init__(**kwargs)
+        self.seed_generator = keras.random.SeedGenerator(seed)
+
     def call(self, inputs):
         z_mean, z_log_var = inputs
-        batch = tf.shape(z_log_var)[0]
-        dim = tf.shape(z_log_var)[1]
-        epsilon = tf.keras.backend.random_normal(shape=(batch, dim))
-        return z_mean + tf.exp(0.5 * z_log_var) * epsilon
+        batch = ops.shape(z_log_var)[0]
```

Comment:
could do this in one line `batch, dim = ops.shape(z_log_var)` or `batch, dim, _ = ops.shape(z_log_var)` if it's higher rank